elements/pl-dataframe/pl-dataframe.py

```diff
@@ -23,6 +24,12 @@ class DisplayLanguage(Enum):
 SHOW_PYTHON_DEFAULT = True
 
 
+def get_python_string(varname: str, frame: pd.DataFrame) -> str:
+    # FIXME this doesn't work with certain data types (an issue with repr as well).
+    code_string = pprint.pformat(frame.to_dict(), width=500, indent=4)
```

Comment:
I'm a little bit worried about this. If `pprint` isn't supporting what was added in `serialization 2`, that might be a bit problematic. Can you describe a little bit about the data type/edge cases that are problematic?

Reply:
So the issue is that the `repr` values for types like `Timestamp` and `float('nan')` don't output valid Python. I'm not quite sure how to fix this without writing a custom serializer of some type.

Reply:
Do you see a path forward to fixing this? If not, I'm inclined to drop the `FIXME` and just document that limitation.

Reply:
Yeah, I'm not really sure how to fix this, even using just repr has this same problem. It's not a deal breaker because it's fixable by the end user, so just adding a note in the docs is probably the way to go.

Reply:
Added to the docs that some types may not be deserialized correctly.

Reply:
Eh, I would say it's a deal breaker as those were two reasons for `v2`. I can't dig into this at the moment as I'm fighting other issues with workspaces and lecture slides. So, it'll be a couple of weeks at the earliest.

Reply:
Oh I didn't mean we shouldn't still attempt to get this working for the edge cases, but this same issue is present in the current version of the code, and there's no immediate fix that should block this PR.

Reply:
Ah, didn't realize that. Thought the dump worked correctly. Ewk! Alrighty, on with the merging on conquest.

Reply:
Yep, the version in this PR works in more cases than the current code from the testing I did.

Reply:
Should we just disable the `show-python` code tab if we detect a problematic type in the interim?

Reply:
I'm not sure if there's a nice way to get a comprehensive list of everything that causes problems. I only found 2, but they can be fixed by setting certain variables.

Reply:
I managed to fix some of the edge cases by adding some extra imports into the template! We will probably run into others but the current version worked for everything in the examples.